apps/els_lsp/src/els_dt_references.erl

```diff
@@ -96,11 +98,24 @@ find_by_id(Kind, Id) ->
   find_by(Pattern).
 
 -spec find_by(tuple()) -> {ok, [item()]}.
-find_by(Pattern) ->
+find_by(#els_dt_references{id = Id} = Pattern) ->
+  Uris = els_text_search:find_candidate_uris(Id),
+  [begin
+     {ok, Document} = els_utils:lookup_document(Uri),
+     POIs = els_dt_document:pois(Document, [ application
+                                           , behaviour
+                                           , implicit_fun
+                                           , include
+                                           , include_lib
+                                           , type_application
+                                           , import_entry
+                                           ]),
+     [register_reference(Uri, POI) || POI <- POIs]
```

Comment:
do I understand correctly that every time I look up references of an entity, it will call `find_by` and try to insert a lot of references in the ref_db (ets entries might be there already, so some of them might be noop) Why do we store the references in ets then? can't we run the matching on the `POIs` list in place?
Alternatively can we mark Uris, which already registered their references, so we can skip lookup_pois+register_references for them?

Reply:
In my latest version (soon to be pushed), we only do this when expanding the POIs and enforcing a full indexing.

Reply:
My major concern here is that, depending on the pattern, we may end up indexing a lot of files. Need to find a good strategy to reduce the number of candidates significantly.